test/testinput/bufr_query_python_to_ioda_test.py

```diff
@@ -37,65 +36,40 @@ def test_bufr_to_ioda():
+    # Create the global attributes
+    obsspace.write_attr('MyGlobal_str', 'My Global String Data')
+    obsspace.write_attr('MyGlobal_int', [1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
 
     # Create the variables
-    timestamp = g.vars.create('MetaData/Timestamp', ioda.Types.int64, scales=[dim_location], params=pdatetime)
-    timestamp.atts.create('units', ioda.Types.str).writeVector.str(['seconds_since_epoch'])
-    timestamp.atts.create('long_name', ioda.Types.str).writeVector.str(['Timestamp'])
-
-    longitude = g.vars.create('MetaData/Longitude', ioda.Types.float, scales=[dim_location], params=pfloat)
-    longitude.atts.create('valid_range', ioda.Types.float, [2]).writeVector.float([-180, 180])
-    longitude.atts.create('units', ioda.Types.str).writeVector.str(['degrees_east'])
-    longitude.atts.create('long_name', ioda.Types.str).writeVector.str(['Longitude'])
-
-    latitude = g.vars.create('MetaData/Latitude', ioda.Types.float,  scales=[dim_location], params=pfloat)
-    latitude.atts.create('valid_range', ioda.Types.float, [2]).writeVector.float([-90, 90])
-    latitude.atts.create('units', ioda.Types.str).writeVector.str(['degrees_north'])
-    latitude.atts.create('long_name', ioda.Types.str).writeVector.str(['Latitude'])
-
-    tb = g.vars.create('ObsValue/brightnessTemperature', ioda.Types.float, scales=[dim_location, dim_channel], params=pfloat)
-    tb.atts.create('valid_range', ioda.Types.float, [2]).writeVector.float([100, 500])
-    tb.atts.create('units', ioda.Types.str).writeVector.str(['K'])
-    tb.atts.create('long_name', ioda.Types.str).writeVector.str(['ATMS Observed (Uncorrected) Brightness Temperature'])
-
-    # Write the data to the variables
-    timestamp.writeNPArray.int64(datetime.astype(np.int64))
-    longitude.writeNPArray.float(lon.flatten())
-    latitude.writeNPArray.float(lat.flatten())
-    tb.writeNPArray.float(rad.flatten())
+    obsspace.create_var('MetaData/timestamp', dtype=datetime.dtype, fillval=datetime.fill_value) \
+        .write_attr('long_name', 'Timestamp') \
+        .write_data(datetime)
+
+    obsspace.create_var('MetaData/latitude', dtype=lat.dtype, fillval=lat.fill_value) \
+        .write_attr('units', 'degrees_north') \
+        .write_attr('long_name', 'Latitude') \
+        .write_attr('valid_range', [-90.0, 90.0]) \
+        .write_data(lat)
+
+    obsspace.create_var('MetaData/longitude', dtype=lon.dtype, fillval=lon.fill_value) \
+        .write_attr('units', 'degrees_east') \
+        .write_attr('long_name', 'Longitude') \
+        .write_attr('valid_range', [-180.0, 180.0]) \
+        .write_data(lon)
+
+    obsspace.create_var('ObsValue/brightnessTemperature',  dtype=rad.dtype,
+                        dim_list=['Location', 'Channel'], fillval=rad.fill_value) \
+        .write_attr('units', 'K') \
+        .write_attr('long_name', 'ATMS Observed (Uncorrected) Brightness Temperature') \
+        .write_attr('valid_range', [100.0, 500.0]) \
```

Comment:
thanks for adding this, likely going to add this `valid_range` also the yamls based ATMS conversion it's missing from the `remap` ones @emilyhcliu worked on